pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -613,6 +613,8 @@ func (c *controller) reconcileMachineHealth(ctx context.Context, machine *v1alph
 					LastUpdateTime: metav1.Now(),
 				}
 				cloneDirty = true
+			} else if clone.Status.CurrentStatus.Phase == v1alpha1.MachineCrashLoopBackOff {
+				return machineutils.ShortRetry, fmt.Errorf("node object not yet created for Machine %s", machine.Name)
 			}
```

Comment:
Let's say we don't call `reconcileMachineHealth`, then also we will have a `longRetry` from the machine reconciliation flow https://github.com/gardener/machine-controller-manager/blob/f87348668d02917d100cfa528a6e079cddec0d51/pkg/util/provider/machinecontroller/machine.go#L187

Reply:
Yes but this is the behaviour which used to be there before also
When node is created, then the event for it should again send the machine for reconciliation because now machine obj will be in `Pending` state.

Reply:
It won't be in a `Pending` state. Even if the machine creation is successful, the object will still be in a `CrashLoopBackOff` state. This is because of https://github.com/gardener/machine-controller-manager/blob/f87348668d02917d100cfa528a6e079cddec0d51/pkg/util/provider/machinecontroller/machine.go#L500. We will have to make a change in `triggerCreationFlow` to change the status or add an explicit condition in machine reconciliation flow to have a `shortRetry` for CLBF machines.

Reply:
The reason is [here](https://github.com/gardener/machine-controller-manager/pull/745/files#:~:text=if%20machine.Status.Node%20!%3D%20nodeName%20%7C%7C%20machine.Status.CurrentStatus.Phase%20%3D%3D%20%22%22%20%7B) the condition `status.Node != nodeName` was removed. This condition enabled transition of `CrashLoopBackoff` -> `Pending` if VM creation is successful.

* So we should fix this.
* Kindly also add a test case for the same
* update the docstring for `CrashLoopBackoff` state, explicitly telling that this states means there are no infra resources